src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -321,25 +338,27 @@ private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperConte
                 var attributeValue = attribute.Value;
                 if (string.Equals(attribute.Key, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
                 {
-                    attributeValue = HtmlEncoder.HtmlEncode(
-                        ShouldAddFileVersion() ?
-                            _fileVersionProvider.AddFileVersionToPath(attributeValue) :
-                            attributeValue);
+                    var attributeStringValue = attributeValue as string;
+
+                    // "href" values come from bound attributes and globbing. Must always be non-null strings.
+                    Debug.Assert(attributeStringValue != null);
```

Comment:
If multiple `TagHelper`s run on the `Link` tag and one runs previously and sets the `TagHelperOutput.Attributes["href"]` to `null` or something other than `string` wont this fail?